pkg/netpol/connlist/exposure_analysis.go

```diff
@@ -72,45 +63,72 @@ func xgressExposureListToXgressExposureDataList(xgressExp []*xgressExposure) []X
 }
 
 func (ep *exposedPeer) IsProtectedByIngressNetpols() bool {
-	return ep.isIngressProtected == protected
+	return ep.ingressExposure.isProtected
 }
 
 func (ep *exposedPeer) IngressExposure() []XgressExposureData {
-	return xgressExposureListToXgressExposureDataList(ep.ingressExposure)
+	return xgressExposureListToXgressExposureDataList(ep.ingressExposure.exposureInfo)
 }
 
 func (ep *exposedPeer) IsProtectedByEgressNetpols() bool {
-	return ep.isEgressProtected == protected
+	return ep.egressExposure.isProtected
 }
 
 func (ep *exposedPeer) EgressExposure() []XgressExposureData {
-	return xgressExposureListToXgressExposureDataList(ep.egressExposure)
+	return xgressExposureListToXgressExposureDataList(ep.egressExposure.exposureInfo)
 }
 
 // ----------------------------------------------------
 
-// gets an exposure map and builds ExposedPeer slice;
+// exposureMaps struct contains:
+// 1. ingressExposureMap : entries of peer to its ingress exposure-analysis data; which may be:
+// - the peer is not protected by ingress netpols
+// - the peer is protected by ingress netpols and exposed unsecure to unknown end-points.(exposure-analysis case)
+// 2. egressExposureMap : entries of peer to its exposure-analysis data; i.e.:
+// - the peer is not protected by egress netpols.
+// or - the peer is exposed on egress to unknown end-points
+//
+// a peer that exists only in one map (one direction); means its protected and exposed securely (to known hosts)
+// on the other direction
+
+// buildExposedPeerListFromExposureMaps gets an exposureMaps struct and builds ExposedPeer slice;
 // list of entries of peer and its exposure connections each
-func buildExposedPeerListFromExposureMap(exposuresMap exposureMap) []ExposedPeer {
+func buildExposedPeerListFromExposureMaps(exposureMaps *exposureMaps) []ExposedPeer {
 	res := make([]ExposedPeer, 0)
-	for p, expData := range exposuresMap {
-		ingExp := make([]*xgressExposure, 0)
-		egExp := make([]*xgressExposure, 0)
-		if expData.isIngressProtected == protected {
-			ingExp = append(ingExp, expData.ingressExposure...)
+	// first loop the ingressExposureMap : for each peer, fill its ingress exposure data
+	// and check if it exists also in the egressExposureMap too - get its egress exposure data;
+	// otherwise it is protected safely on egress (add default value)
+	for p, ingressExpData := range exposureMaps.ingressExposureMap {
+		// default value for egress exposure
+		egressExposureData := &peerXgressExposureData{
+			isProtected:  true,
+			exposureInfo: nil,
```

Comment:
the code for generating the default object of `peerXgressExposureData` is duplicated (here and in the second loop), can have instead a function such as `func newDefaultXgressExposureData()  *peerXgressExposureData`

Reply:
done